waffle/tests/test_waffle.py

```diff
@@ -394,3 +397,47 @@ def test_read_from_write_db(self):
             # The next read should now be directed to the write DB, ensuring
             # the cache and DB are in sync.
             assert waffle.sample_is_active(sample.name)
+
+
+class TransactionTests(TransactionTestCase):
+    @unittest.skipIf('sqlite3' in settings.DATABASES['default']['ENGINE'],
+                     'This test uses threads, which the sqlite3 DB engine '
+                     'does not support.')
+    def test_update_switch_in_transaction(self):
+        """Wait to invalidate the cache until after the current transaction."""
```

Comment:
I appreciate all the comments here, this is definitely not an obvious issue.

Would you mind adding a slightly longer description to the doc block (as a second paragraph) and include a link to #296?

Should we have similar tests for `Flag` and `Sample`? I know the behavior exists in `BaseModel` but that's an implementation detail, not a public API. If we needed to refactor it, especially in support of extensible models, it could change or break.

Reply:
For sure, I can add another comment explicitly referencing #296.

I'm also happy to add equivalent tests for `Flag` and `Sample`.